addon/mixins/route.js

```diff
@@ -236,42 +251,83 @@ const RouteMixin = Mixin.create({
     @private
     @method _infinityLoad
     @param {Ember.ArrayProxy} infinityModel
+    @param {Integer} increment - to increase page by 1 or -1
    */
-  _infinityLoad(infinityModel) {
+  _infinityLoad(infinityModel, increment) {
     if (get(infinityModel, '_loadingMore') || !get(infinityModel, '_canLoadMore')) {
       return;
     }
 
-    this._loadNextPage(infinityModel);
+    this._loadNextPage(infinityModel, increment);
   },
 
   /**
     load the next page from the adapter and update the model
+    set current height of elements.  If loadPrevious, we will use this value to scroll back down the page
 
     @private
     @method _loadNextPage
     @param {Ember.ArrayProxy} infinityModel
+    @param {Integer} increment - to increase page by 1 or -1. Default to increase by one page
     @return {Ember.RSVP.Promise} A Promise that resolves the model
    */
-  _loadNextPage(infinityModel) {
+  _loadNextPage(infinityModel, increment = 1) {
     set(infinityModel, '_loadingMore', true);
+    set(this, '_previousScrollHeight', this._calculateHeight(infinityModel));
 
     const modelName = get(infinityModel, '_infinityModelName');
-    const params    = infinityModel.buildParams();
+    const params    = infinityModel.buildParams(increment);
 
     return this._requestNextPage(modelName, params)
       .then(newObjects => this._afterInfinityModel(this)(newObjects, infinityModel))
       .then(newObjects => this._doUpdate(newObjects, infinityModel))
       .then(infinityModel => {
-        infinityModel.incrementProperty('currentPage');
+        if (increment === 1) {
+          // scroll down to load next page
+          infinityModel.incrementProperty('currentPage');
+        } else {
+          let viewportElem = get(infinityModel, '_scrollable') ? document.querySelector(get(infinityModel, '_scrollable')) : window;
+          scheduleOnce('afterRender', this, '_updateScrollTop', { infinityModel, viewportElem });
```

Comment:
if `increment === -1`, then we are going backwards.  So we need to scroll back down the page `afterRender`.